src/lib.rs

```diff
@@ -36,13 +36,36 @@
 
 macro_rules! resultify {
     ($ret:expr) => {
-        match $ret >= 0 {
-            true    => Ok($ret as _),
-            false   => Err(io::Error::from_raw_os_error(-$ret)),
+        match $ret {
+            ret if ret >= 0 => Ok(ret as _),
+            ret => Err(io::Error::from_raw_os_error(-ret)),
         }
     }
 }
 
+#[test]
```

Comment:
Could you put this in a tests module marked `#[cfg(test)]`?

Reply:
Will do.